src/callbacks.c

```diff
@@ -327,14 +327,27 @@ void on_open1_activate(GtkMenuItem *menuitem, gpointer user_data)
 }
 
 
+static void reloadDoc(GeanyDocument *doc)
+{
+	g_return_if_fail(doc != NULL);
+
+	document_reload_prompt(doc, NULL);
+}
+
+
 /* reload file */
 void on_toolbutton_reload_clicked(GtkAction *action, gpointer user_data)
 {
-	GeanyDocument *doc = document_get_current();
+	reloadDoc(document_get_current());
+}
 
-	g_return_if_fail(doc != NULL);
 
-	document_reload_prompt(doc, NULL);
+/* reload all file */
+void on_toolbutton_reload_all_clicked(GtkAction *action, gpointer user_data)
+{
+	for (guint doc_count = 0; doc_count < documents_array->len; ++doc_count) {
```

Comment:
I don’t know if this is relevant, but the [HACKING](https://github.com/geany/geany/blob/master/HACKING) docs say:

> Bugs to watch out for [...] Forgetting to check *doc->is_valid* when looping through *documents_array* - instead use *foreach_document()*.